### PR TITLE
fix name_or_path usage in HF save/load usage

### PR DESCRIPTION
**Pull request by dakinggg**
# What does this PR do?
This PR fixes the usage of `name_or_path` in the saving/loading of the HuggingFace tokenizer to be resilient to the key not existing, as apparently it was removed in a minor release. It also pins transformers so this does not happen again.

# What issue(s) does this change relate to?
Closes [CO-1942](https://mosaicml.atlassian.net/browse/CO-1942)

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [x] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [x] Did you update any related docs and document your change?
- [x] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

[CO-1942]: https://mosaicml.atlassian.net/browse/CO-1942?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/datasets/test_in_context_learning_datasets.py [hunk 1]
```diff
@@ -246,20 +247,23 @@ def test_lm_task_evaluation(device, dataset_uri, num_fewshot, tiny_gpt2_tokenize
 
 
 @pytest.mark.parametrize('dataset_uri', ['piqa_small.jsonl', 'hellaswag_small.jsonl'])
-@device('gpu')
+# @device('cpu')
```
[the remark this answers is not in the document]
Comment (dakinggg) [reply tests/datasets/test_in_context_learning_datasets.py:250]: yeah woops, wasn't quite ready for review, thanks!

File: tests/datasets/test_in_context_learning_datasets.py [hunk 2]
```diff
@@ -270,10 +274,10 @@ def test_mc_task_evaluation(device, num_fewshot, dataset_uri, tiny_gpt2_tokenize
 
     evaluator = Evaluator(label='lambada', dataloader=dl, metric_names=['InContextLearningMultipleChoiceAccuracy'])
 
-    config = transformers.AutoConfig.from_pretrained('EleutherAI/gpt-neo-125M')
-    model = transformers.AutoModelForCausalLM.from_config(config)
+    # config = transformers.AutoConfig.from_pretrained('EleutherAI/gpt-neo-125M')
+    # model = transformers.AutoModelForCausalLM.from_config(config)
```
[the remark this answers is not in the document]
Comment (dakinggg) [reply tests/datasets/test_in_context_learning_datasets.py:278]: yup, thanks!

**Comment by dakinggg** [comment]
@nik-mosaic mind rereviewing? cleaned it up

File: tests/models/test_hf_model.py [hunk 3]
```diff
@@ -104,6 +104,13 @@ def check_hf_tokenizer_equivalence(tokenizer1, tokenizer2):
     tokenizer1.__dict__.pop('deprecation_warnings')
     tokenizer2.__dict__.pop('deprecation_warnings')
 
+    # name or path will be the path that the tokenizer was loaded from, which will just be a temporary directory for
+    # the reloaded tokenizer
+    tokenizer1.__dict__.pop('name_or_path')
+    tokenizer2.__dict__.pop('name_or_path')
+    tokenizer1.init_kwargs.pop('name_or_path', None)
+    tokenizer2.init_kwargs.pop('name_or_path', None)
```
Comment (mvpatel2000) [inline tests/models/test_hf_model.py:112]: I don't really understand this comment. Can you indicate where we re-set this var or why its not needed?
```suggestion
    # Name or path will be the path that the tokenizer was loaded from, which will just be a temporary directory for
    # the reloaded tokenizer
    tokenizer1.__dict__.pop('name_or_path')
    tokenizer2.__dict__.pop('name_or_path')
    tokenizer1.init_kwargs.pop('name_or_path', None)
    tokenizer2.init_kwargs.pop('name_or_path', None)
```
Nit: capitalize start of comment

**Review by mvpatel2000 — APPROVED** [review]

**Pull request merged**